langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py

```diff
@@ -85,19 +240,28 @@ def run(configuration, agent=None, max_loops=-1):
         agent = init_agent(configuration)
 
     if hasattr(agent, 'read'):
-        source = agent
+        source = RuntimeSource(agent, agent_id, agent_type)
     else:
-        source = TopicConsumerWithDLQSource(consumer, dlq_producer if dlq_producer else NoopTopicProducer())
+        source = RuntimeSource(
+            TopicConsumerWithDLQSource(consumer, dlq_producer if dlq_producer else NoopTopicProducer()),
+            'topic-source',
+            'topic-source')
+
+    agent_info.source = source
 
     if hasattr(agent, 'write'):
-        sink = agent
+        sink = RuntimeSink(agent, agent_id, agent_type)
     else:
-        sink = TopicProducerSink(producer)
+        sink = RuntimeSink(TopicProducerSink(producer), 'topic-sink', 'topic-sink')
```

Comment:
It's the same agent-id/agent-type for Java topic producer.
Shoould we use different names ?

Reply:
it's perfect, the UI may expect those names